transfers/utils.py

```diff
@@ -130,65 +129,69 @@ def activate_endpoint(tc, ep):
         )
         input("Press ENTER after activating the endpoint:")
         r = tc.endpoint_autoactivate(ep, if_expires_in=3600)
-        
+
     return
 
 
 def get_contents(ep, path):
     """Get the contents of a Globus directory as a list of string values"""
     ep_path = f"{ep}:{path}"
-    
+
     command = ["globus", "ls", ep_path]
     try:
         out = subprocess.check_output(command, stderr=subprocess.STDOUT)
     except subprocess.CalledProcessError as exc:
         out_list = re.split("\n|\s\s\s\s\s", exc.output.decode("utf-8"))
-        
+
         try:
             http_status = out_list[out_list.index("HTTP status:") + 1]
         except ValueError:
             print(exc.output.decode("utf-8"))
             print(command)
             exit()
         return int(http_status)
-        
+
     contents = [c for c in out.decode("utf-8").split("\n") if c != ""]
-    
+
     return contents
 
 
 def operation_ls(tc, ep, path):
     """ls operation on the specified endpoint:path. Wrapper for the globus_sdk.services.transfer.client.TransferClient.operation_ls function.
-    
+
     Args:
         tc (globus_sdk.services.transfer.client.TransferClient): transfer client object
         ep (str): endpoint ID
         path (str): path on the specified endpoint to run ls
-    
+
     Returns:
         contents (list): list of contents (as strings) in the directory or HTTP error code
     """
-    try:
-        r = tc.operation_ls(ep, path)
-    except globus_sdk.TransferAPIError as exc:
-        return exc.http_status
-        
+    n = 5
+    for retry in range(n):
+        try:
+            r = tc.operation_ls(ep, path)
+        except globus_sdk.TransferAPIError as exc:
+            if retry < n - 1:
+                continue
+            else:
+                return exc.http_status
+
```

Comment:
I think we actually might want to keep this as is. I added an exponential time backoff to the call of this function should it return specific status. I think I was still getting errors even with the retries because this would retry immediately. I think it is kind of helpful to keep this function simply "try it once, and return the error code if it fails" and we can add wrappers for the retries like I did in `esgf_holdings.py`. Open to discussing though. 

```suggestion
    try:
        r = tc.operation_ls(ep, path)
    except globus_sdk.TransferAPIError as exc:
        return exc.http_status
```

Reply:
revised in [c5f3c2b](https://github.com/ua-snap/cmip6-utils/pull/54/commits/c5f3c2bf8d5266fc85a4132006b8c9ccc5e5e325)